x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -141,11 +145,17 @@ public void syntaxError(
             String message,
             RecognitionException e
         ) {
-            if (recognizer instanceof EsqlBaseParser parser && parser.isDevVersion() == false) {
-                Matcher m = REPLACE_DEV.matcher(message);
-                message = m.replaceAll(StringUtils.EMPTY);
+            if (recognizer instanceof EsqlBaseParser parser) {
+                Matcher m = REPLACE_LP.matcher(message);
+                message = m.replaceAll(BEFORE_REPLACEMENT + "(" + AFTER_REPLACEMENT);
+                m = REPLACE_LEFT_BRACKET.matcher(message);
+                message = m.replaceAll(BEFORE_REPLACEMENT + "[" + AFTER_REPLACEMENT);
```

Comment:
Is there a chance we could introduce corresponding tokens to the mode back?

I think it is a bit better to generate correct message from antlr when possible rather than rewrite it in our code.
This also introduces a string usage of the token names outside of the grammar and one more place to update if that is ever changed or renamed.